lib/middleman-images/image.rb

```diff
@@ -1,32 +1,32 @@
 module Middleman
   module Images
     class Image < Middleman::Sitemap::Resource
-      attr_reader :destination, :source
-
       IGNORE_RESIZING = {
         ".svg" =>  "WARNING: We did not resize %{file}. Resizing SVG files will lead to ImageMagick creating an SVG with an embedded binary image thus making the file way bigger.",
         ".gif" =>  "WARNING: We did not resize %{file}. Resizing GIF files will remove the animation. If your GIF file is not animated, use JPG or PNG instead.",
       }.freeze
 
-      def initialize(app, source, destination, options = {})
-        @app = app
-        @source = source
-        @destination = destination
-        @options = options
-        @cache = File.join(app.root, options[:cache_dir], destination).to_s
+      attr_reader :app, :original_source_file
+
+      def initialize(store, path, source, options = {})
+        @original_source_file = source
+
+        cache = File.join(store.app.root, options.delete(:cache_dir), path).to_s
         FileUtils.mkdir_p File.dirname(cache)
 
-        super(app.sitemap, destination, cache)
+        @processing_options = options
+
+        super(store, path, cache)
```

Comment:
Mich hat gerade (zu) lange, die Benennung der `cache` Variable verwirrt. Aus zwei Gründen: ersteinmal war mir aus Zeile 14 nicht ganz klar, ob es sich hier um einen Ordner oder schon um die Datei selbst handeln soll. Zweitens war mir der Zusammenhang zur Methode `source_file` (die in diesem Fall ja auf die gleiche Datei zeigen soll) nicht klar.
Falls diese Probleme nachvollziehbar sind, würde ich deshalb für einen anderen Namen plädieren. Zum Beispiel `processed_source_file` oder so? (Namen sind schwierig.)

Reply:
Guter Punkt. Ich war mir hier auch nicht sicher, ob das klar wird.